### PR TITLE
Migration to .NET 6

### PR DESCRIPTION
**Pull request by Piotrreek**

### CODE REVIEW

**Comment by tomaszmadeyski** [comment]
A general comment: we should not commit the whole `modules/*` subtree - this will be installed at the project that installs this nuget as its dependency

**Comment by tomaszmadeyski** [comment]
`upgrade-assistant.clef` is also leftover after upgrade process - we don't need it

File: Styleguide.EPiServer/ContainerExtensions.cs [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+using System.Collections.Generic;
+using System.IO;
 using EPiServer;
 using EPiServer.Construction;
 using EPiServer.DataAbstraction;
 using Forte.Styleguide.EPiServer.ContentProvider;
 using Forte.Styleguide.EPiServer.JsonConverters;
 using Newtonsoft.Json;
 using StructureMap;
+using Forte.Styleguide.EPiServer;
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ContainerExtensions.cs]: we have some redundant usings here

File: Styleguide.EPiServer/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -13,6 +14,7 @@
 [assembly: AssemblyCopyright("Copyright ©  2018")]
 [assembly: AssemblyTrademark("")]
 [assembly: AssemblyCulture("")]
+//[assembly: AspMvcViewLocationFormat(@"..\..\Styleguide\Views\{1}\{0}.cshtml")]
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/Properties/AssemblyInfo.cs]: if not relevant let's just remove it (together with redundant usings)

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            services.AddSingleton<IViewCompilerProvider, ModuleViewCompilerProvider>();
+            services.AddScoped(typeof(ComponentCatalogLoader));
+
+            StyleguideServiceColletionExtensions.AddStyleGuide(services);
+            return services;
+        }
+
+        public static void UseEpiServerStyleguide(this IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            AppDomain.CurrentDomain.SetData("ContentRootPath", env.ContentRootPath);
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: I think we should move away from this way of storing configuration data (AppDomain...)
In .net core common practice is to use options pattern: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-6.0

File: Styleguide.EPiServer/ContainerExtensions.cs [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
 
 namespace Forte.Styleguide.EPiServer
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ContainerExtensions.cs]: General comment: we should move away from StructureMap package completely and assume we use built in IServiceCollection as a DI container, so this class is not relevant - we should redo it with the use of IServiceCollection

File: Styleguide.EPiServer/ContentProvider/StyleguideContentProvider.cs [hunk 5]
```diff
@@ -30,8 +30,8 @@ private static StyleguideContentProvider Install()
 
             provider.Initialize(ProviderName, new NameValueCollection
             {
-                { ContentProviderElement.EntryPointString, entryPoint.ToString() },
-                { ContentProviderElement.CapabilitiesString, ContentProviderCapabilities.None.ToString() }
+                { "entryPoint", entryPoint.ToString() },
+                {"capabilities", ContentProviderCapabilities.None.ToString() }
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ContentProvider/StyleguideContentProvider.cs]: inconsistent spacing

File: Styleguide.EPiServer/HttpContextHelper.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using System.Web;
+#endif
+
+namespace Forte.Styleguide.EPiServer
+{
+    /// <summary>
+    /// Temporary helper class for retrieving the current <see cref="HttpContext"/> . This temporary
+    /// workaround should be removed in the future and <see cref="HttpContext"/> should be retrieved
+    /// from the current controller, middleware, or page instead.
+#if NET || NETCOREAPP
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/HttpContextHelper.cs]: I wonder: do we need all these `ifs` once we're targetting .net6 only?

Comment (tomaszmadeyski) [reply Styleguide.EPiServer/HttpContextHelper.cs]: plus: this class does't seem to be used anywhere?

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: redundant usings

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace Forte.Styleguide.EPiServer
+{
+
+    public static class ServiceColletionExtensions
+    {
+        public static IServiceCollection AddStyleGuide(this IServiceCollection services)
+        {
+            services.AddControllersWithViews();
+            ServiceDescriptor descriptor = services.FirstOrDefault(s => s.ServiceType == typeof(IViewCompilerProvider));
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: var

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Forte.Styleguide.EPiServer
+{
+
+    public static class ServiceColletionExtensions
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: a typo

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        public static IServiceCollection AddStyleGuide(this IServiceCollection services)
+        {
+            services.AddControllersWithViews();
+            ServiceDescriptor descriptor = services.FirstOrDefault(s => s.ServiceType == typeof(IViewCompilerProvider));
+            services.Remove(descriptor);
+            services.AddSingleton<IViewCompilerProvider, ModuleViewCompilerProvider>();
+            services.AddScoped(typeof(ComponentCatalogLoader));
+
+            StyleguideServiceColletionExtensions.AddStyleGuide(services);
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: I think it would be better to call these methods differently (ie `AddStyleGuideEpiserver` and `AddStyleGuideCore`), then you can call simply services.AddStyleGuideCOre here

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+            return services;
+        }
+
+        public static void UseEpiServerStyleguide(this IApplicationBuilder app, IWebHostEnvironment env)
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: inconsistency when it comes to naming - AddStyleGuide but UseEpiServerStyleguide - we should keep it consistent - UseX and AddX

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            AppDomain.CurrentDomain.SetData("ContentRootPath", env.ContentRootPath);
+            AppDomain.CurrentDomain.SetData("WebRootPath", env.WebRootPath);
+            StyleguideServiceColletionExtensions.UseEpiServerStyleguide(app, env);
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: same here with naming - let's distinct these when it comes to names

File: Styleguide.Tests/Styleguide.Tests.csproj [hunk 13]
```diff
@@ -1,81 +1,22 @@
...
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\Styleguide\Styleguide.csproj">
-      <Project>{84ebe80b-8bbe-43c5-a7ef-6cc22b4c14fa}</Project>
-      <Name>Styleguide</Name>
-    </ProjectReference>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.2.0" />
+    <PackageReference Include="NUnit" Version="3.11.0" />
+    <PackageReference Include="NUnit3TestAdapter" Version="4.2.1" />
+    <PackageReference Include="Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers" Version="0.3.330701">
```
Comment (tomaszmadeyski) [inline Styleguide.Tests/Styleguide.Tests.csproj]: I think we don't need this one

**Comment by tomaszmadeyski** [comment]
a global comment: I see you're not adding empty line at the end of the file, which is considered as a good practice. I think there should be a seeting for that in your IDE (there's one in Rider for sure)

File: Styleguide/Controllers/StyleGuideController.cs [hunk 14]
```diff
@@ -1,70 +1,69 @@
...
         public ActionResult Index()
         {
-            var catalog = this.loader.Load(reload: true);
+            
+            var catalog = _loader.Load(reload: true);
             var model = new StyleguideIndexViewModel(catalog.Components);
 
-            return View(model);
+            
+            return View("~/Areas/MyFeature/Pages/Index.cshtml", model);
```
Comment (tomaszmadeyski) [inline Styleguide/Controllers/StyleGuideController.cs]: can we make this path to look less like copied from tutorial ? 😉

File: Styleguide/Controllers/StyleGuideController.cs [hunk 15]
```diff
@@ -1,70 +1,69 @@
 ﻿using System.Net;
 using System.Threading.Tasks;
```
Comment (tomaszmadeyski) [inline Styleguide/Controllers/StyleGuideController.cs]: some redundant usings

File: Styleguide/Controllers/StyleGuideController.cs [hunk 16]
```diff
@@ -1,70 +1,69 @@
...
-using System.Web.Mvc;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
 
 namespace Forte.Styleguide
```
Comment (tomaszmadeyski) [inline Styleguide/Controllers/StyleGuideController.cs]: can we adjust namespace so it reflects folder structure? (Controllers part missing)

File: Styleguide/ServiceCollectionExtensions.cs [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Forte.Styleguide
+{
+    public static class StyleguideServiceColletionExtensions
```
Comment (tomaszmadeyski) [inline Styleguide/ServiceCollectionExtensions.cs]: typo

File: Styleguide/Styleguide.csproj [hunk 18]
```diff
@@ -1,126 +1,30 @@
...
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{84EBE80B-8BBE-43C5-A7EF-6CC22B4C14FA}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Forte.Styleguide</RootNamespace>
-    <AssemblyName>Forte.Styleguide</AssemblyName>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net6.0</TargetFramework>
+	  <RootNamespace>Forte.Styleguide</RootNamespace>
```
Comment (tomaszmadeyski) [inline Styleguide/Styleguide.csproj:10]: can we unify indentation?

File: Styleguide/Styleguide.csproj [hunk 19]
```diff
@@ -1,126 +1,30 @@
...
+	<ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
   </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-</Project>
\ No newline at end of file
+	<ItemGroup>
+		<PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+		<PackageReference Include="System.Data.DataSetExtensions" Version="4.5.0" />
+		<PackageReference Include="Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers" Version="0.3.330701">
```
Comment (tomaszmadeyski) [inline Styleguide/Styleguide.csproj]: I think we don't need this one

File: Styleguide/Styleguide.csproj [hunk 20]
```diff
@@ -1,126 +1,30 @@
...
+			<PrivateAssets>all</PrivateAssets>
+		</PackageReference>
+	</ItemGroup>
+	<ItemGroup>
+		<PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
```
Comment (tomaszmadeyski) [inline Styleguide/Styleguide.csproj]: When defining dependencies let's try to define version as a range - ie `Version="[10.0.0,14)"` This way we're not forcing end user to specific version of a package - especially that Newtonsoft.Json in versions older than 13.0.1 has a known security issue

File: Styleguide/Areas/MyFeature/Pages/Index.cshtml [hunk 21]
```diff
@@ -8,7 +7,7 @@
     <title>Style Guide</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <script type="text/javascript" src="https://code.jquery.com/jquery-3.3.1.js" integrity="sha256-2Kok7MbOyxpgUVvAk/HJ2jigOSYS2auK4Pfzbm7uH60="crossorigin="anonymous"></script>
```
Comment (tomaszmadeyski) [inline Styleguide/Areas/MyFeature/Pages/Index.cshtml]: can we quit using jquery ? it seems there's no rocket science here and we can replace it with vanilla js

File: Styleguide/Areas/MyFeature/Pages/Index.cshtml [hunk 22]
```diff
@@ -87,7 +86,7 @@
         </div>
     </div>
 </div>
-<script type="text/javascript" src="~/Static/Styleguide/js/mandelbrot.js" ></script>
+<script type="text/javascript" src="_content/Styleguide/Styleguide/js/mandelbrot.js" ></script>
```
Comment (tomaszmadeyski) [inline Styleguide/Areas/MyFeature/Pages/Index.cshtml]: hmm, that's odd - is `_content` some sort of RazorClassLibrary pattern?

**Review by tomaszmadeyski — CHANGES_REQUESTED** [review]

File: Styleguide/Areas/MyFeature/Pages/Index.cshtml (hunk 22 above)
Comment (Piotrreek) [reply Styleguide/Areas/MyFeature/Pages/Index.cshtml]: yes, I found this in MS docs

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+
+namespace Forte.Styleguide.EPiServer
+{
+
+    public static class MyServer
+    {
+        public static string MapPath(string path)
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: seems like we don't need this class/method at all - can be removed

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+            return path;
+            //return Path.Combine(
+            //    (string)AppDomain.CurrentDomain.GetData("ContentRootPath"),
+            //    path);
+        }
+    }
+
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddStyleGuideEPiServer(this IServiceCollection services, string featuresRootPath = "Features", string componentFileNameExtension = ".styleguide.json", string layoutPath = null)
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: EpiServer in the method name (lower `p`)

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 25]
```diff
@@ -0,0 +1,76 @@
...
+                            provider.GetRequiredService<ContentAreaConverter>()
+                        }
+                    },
+                    provider.GetRequiredService<IViewEngine>(),
+                    layoutPath));
+
+            return services;
+        }
+
+        public static void UseStyleguideEPiServer(this IApplicationBuilder app, IWebHostEnvironment env)
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: if this does not do anything - let's remove it

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 26]
```diff
@@ -0,0 +1,76 @@
...
+
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddStyleGuideEPiServer(this IServiceCollection services, string featuresRootPath = "Features", string componentFileNameExtension = ".styleguide.json", string layoutPath = null)
+        {
+            services.AddControllersWithViews();
+            var descriptor = services.FirstOrDefault(s => s.ServiceType == typeof(IViewCompilerProvider));
+            services.Remove(descriptor);
+            services.AddSingleton<IViewCompilerProvider, ModuleViewCompilerProvider>();
+            services.AddScoped(typeof(ComponentCatalogLoader));
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: I think it's safer to use `AddTransient` instead of `AddScoped`

File: Styleguide.EPiServer/Styleguide.EPiServer.csproj [hunk 27]
```diff
@@ -1,197 +1,62 @@
...
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/Styleguide.EPiServer.csproj:8]: can we target both net5 and net6?

File: Styleguide/ResultConverter.cs [hunk 28]
```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Forte.Styleguide
+{
+    class ResultConverter : JsonConverter
```
Comment (tomaszmadeyski) [inline Styleguide/ResultConverter.cs]: can we add `internal` explicitly?

File: Styleguide/ResultConverter.cs [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return (objectType == typeof(ViewDataDictionary));
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            JObject jo = JObject.Load(reader);
```
Comment (tomaszmadeyski) [inline Styleguide/ResultConverter.cs]: var ?

File: Styleguide/Views/Styleguide/Index.cshtml [hunk 30]
```diff
@@ -1,14 +1,12 @@
...
+﻿@model Forte.Styleguide.Views.Styleguide.StyleguideIndexViewModel
 
 <!DOCTYPE html>
 <html class="has-js" dir="ltr">
 <head>
     <title>Style Guide</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <script type="text/javascript" src="https://code.jquery.com/jquery-3.3.1.js" integrity="sha256-2Kok7MbOyxpgUVvAk/HJ2jigOSYS2auK4Pfzbm7uH60="crossorigin="anonymous"></script>
-    <link rel="stylesheet" type="text/css" href="~/Static/Styleguide/css/white.css"/>
+    @*<script type="text/javascript" src="https://code.jquery.com/jquery-3.3.1.js" integrity="sha256-2Kok7MbOyxpgUVvAk/HJ2jigOSYS2auK4Pfzbm7uH60="crossorigin="anonymous"></script>*@
```
Comment (tomaszmadeyski) [inline Styleguide/Views/Styleguide/Index.cshtml]: once unused it can be removed

File: Styleguide/Views/Styleguide/Index.cshtml [hunk 31]
```diff
@@ -87,31 +85,35 @@
...
-            
-            $("iframe#component").attr("src", "styleguide/Component/" + componentName);
-            $("iframe#componentContext").attr("src", "styleguide/ComponentContext/" + componentName);
-        });
+        document.querySelectorAll('a.Tree-entityLink').forEach(x => x.addEventListener('click',
+            (e) => {
+                e.preventDefault();
+
+                var componentName = e.target.getAttribute('data-component')
+                console.log(e);
```
Comment (tomaszmadeyski) [inline Styleguide/Views/Styleguide/Index.cshtml]: leftover?

File: Styleguide/ModuleViewCompiler.cs [hunk 32]
```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections.Concurrent;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+
+namespace Forte.Styleguide
+{
+    public class ModuleViewCompiler : IViewCompiler
```
Comment (tomaszmadeyski) [inline Styleguide/ModuleViewCompiler.cs:12]: a general comment - can we use `var` instead of concrete type wherever it's possible (in this class)

Comment (tomaszmadeyski) [reply Styleguide/ModuleViewCompiler.cs:12]: and a question - did you write this class from scratch or it is port of some old class that existed before?

**Review by tomaszmadeyski — CHANGES_REQUESTED** [review]

File: Styleguide/ModuleViewCompiler.cs (hunk 32 above)
Comment (Piotrreek) [reply Styleguide/ModuleViewCompiler.cs:12]: I found this class in internet because a replacement for compiling views was necessary

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 33]
```diff
@@ -0,0 +1,56 @@
...
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddStyleGuideEpiServer(this IServiceCollection services, string featuresRootPath = "Features", string componentFileNameExtension = ".styleguide.json", string layoutPath = null)
+        {
+            services.AddControllersWithViews();
+            var descriptor = services.FirstOrDefault(s => s.ServiceType == typeof(IViewCompilerProvider));
+            services.Remove(descriptor);
+            services.AddSingleton<IViewCompilerProvider, ModuleViewCompilerProvider>();
+            services.AddTransient(typeof(ComponentCatalogLoader));
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs]: the more common approach is to use generic version of AddTransient - `services.AddTransient<ComponentCatalogLoader>();`

File: Styleguide/Controllers/StyleGuideController.cs [hunk 34]
```diff
@@ -1,70 +1,69 @@
...
-using System.Threading.Tasks;
-using System.Web.Mvc;
+﻿using Forte.Styleguide;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Styleguide.Views.Styleguide;
 
-namespace Forte.Styleguide
+namespace Styleguide.Controllers
```
Comment (tomaszmadeyski) [inline Styleguide/Controllers/StyleGuideController.cs]: we should stick to Forte.Styleguide (as defined in the csproj file)

File: Styleguide/MvcPartialComponentDescriptor.cs [hunk 35]
```diff
@@ -75,9 +72,9 @@ private static Type ResolveViewModelType(BuildManagerCompiledView view)
             return typeof(object);
         }
 
-        private static RazorView FindPartialView(ControllerContext context, string name)
+        private RazorView FindPartialView(ControllerContext context, string name)
         {
-            var result = ViewEngines.Engines.FindPartialView(context, name);
+            var result = _engine.FindView(context, name, false);
```
Comment (tomaszmadeyski) [inline Styleguide/MvcPartialComponentDescriptor.cs:78]: I'm bit concerned about change FindPartialView -> FindView. VIew and Partial view are two different things, let's test how it works

File: Styleguide/Views/Styleguide/MvcPartialComponent.cshtml [hunk 36]
```diff
@@ -1,7 +1,5 @@
-﻿@using System.Web.Mvc
-@using System.Web.Mvc.Html
 
-@model Forte.Styleguide.MvcPartialComponentViewModel
+@model Styleguide.Views.Styleguide.MvcPartialComponentViewModel
```
Comment (tomaszmadeyski) [inline Styleguide/Views/Styleguide/MvcPartialComponent.cshtml]: again, invalid namespace, should start with Forte. Please recheck it globally

File: Styleguide/MvcPartialComponentDescriptor.cs (hunk 35 above)
Comment (Piotrreek) [reply Styleguide/MvcPartialComponentDescriptor.cs:78]: I did not find better replacement for that. So we have to test it and if it doen't work we will have to do reserach

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs [hunk 37]
```diff
@@ -0,0 +1,58 @@
...
+        public static IServiceCollection AddStyleGuideEpiServer(this IServiceCollection services, string featuresRootPath = "Features", string componentFileNameExtension = ".styleguide.json", string layoutPath = null)
+        {
+            var descriptor = services.FirstOrDefault(s => s.ServiceType == typeof(IViewCompilerProvider));
+            services.Remove(descriptor);
+            services.AddSingleton<IViewCompilerProvider, ModuleViewCompilerProvider>();
+            services.AddTransient<ComponentCatalogLoader>();
+            services.AddTransient<ContentConverter>();
+            services.AddTransient<ContentReferenceConverter>();
+            services.AddTransient<ContentAreaConverter>();
+            services.AddScoped<IViewEngine, RazorViewEngine>();
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/ServiceCollectionExtensions.cs:29]: I wonder do we need this registration? Documentation of RazorViewEngine says it is default implementation of IViewEngine so maybe it is registered by Mvc by default?

File: Styleguide/Views/Styleguide/MvcPartialComponent.cshtml [hunk 38]
```diff
@@ -1,11 +1,8 @@
-﻿@using System.Web.Mvc
-@using System.Web.Mvc.Html
+@using Forte.Styleguide.Views.Styleguide
```
Comment (tomaszmadeyski) [inline Styleguide/Views/Styleguide/MvcPartialComponent.cshtml:1]: can't we do as before with model:

`@model MvcPartialComponentViewModel`

and then use 
`Model.LayoutPath`

instead of 
`(MvcPartialComponentViewModel)ViewData["Model"])`
?

File: Styleguide/Views/Styleguide/MvcPartialComponentContext.cshtml [hunk 39]
```diff
@@ -9,10 +7,8 @@
     <body>
         <div class="Browser-panel Browser-code is-active" data-role="tab-panel">
             <code class="Code Code--lang-json">
-                <pre>@Model.Context</pre>
+                <pre>@(((MvcPartialComponentContextViewModel)ViewData["Model"]).Context)</pre>
```
Comment (tomaszmadeyski) [inline Styleguide/Views/Styleguide/MvcPartialComponentContext.cshtml:10]: same here with @model

File: Styleguide.EPiServer/Styleguide.EPiServer.csproj [hunk 40]
```diff
@@ -1,197 +1,63 @@
...
-    <Reference Include="StructureMap, Version=4.6.1.0, Culture=neutral, PublicKeyToken=null">
-      <HintPath>..\packages\StructureMap.4.6.1\lib\net45\StructureMap.dll</HintPath>
-    </Reference>
-    <Reference Include="StructureMap.Web, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\structuremap.web.4.0.0.315\lib\net40\StructureMap.Web.dll</HintPath>
-    </Reference>
-    <Reference Include="System" />
-    <Reference Include="System.ComponentModel.Annotations, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\packages\System.ComponentModel.Annotations.4.4.0\lib\net461\System.ComponentModel.Annotations.dll</HintPath>
+    <None Remove="C:\Users\PiotrGebalski\.nuget\packages\episerver.cms.tinymce\3.2.0\contentFiles\any\any\modules\_protected\EPiServer.Cms.TinyMce\EPiServer.Cms.TinyMce.zip" />
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/Styleguide.EPiServer.csproj]: this seems irrelevant ;)

Comment (tomaszmadeyski) [reply Styleguide.EPiServer/Styleguide.EPiServer.csproj]: (I mean whole this part with c:\users)

File: Styleguide.EPiServer/Styleguide.EPiServer.csproj [hunk 41]
```diff
@@ -1,197 +1,63 @@
...
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="6.0.0" />
+    <PackageReference Include="OdeToCode.AddFeatureFolders" Version="2.0.3" />
+    <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
+    <PackageReference Include="System.Data.DataSetExtensions" Version="4.5.0" />
+    <PackageReference Include="System.Threading.Tasks.Dataflow" Version="6.0.0" />
+    <PackageReference Include="System.Configuration.ConfigurationManager" Version="6.0.0" />
   </ItemGroup>
   <ItemGroup>
-    <None Include="packages.config" />
+	  <PackageReference Include="EPiServer.Commerce" Version="14.5.0" />
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/Styleguide.EPiServer.csproj]: can we:
1. remove EpiServer.Commerce 
2. specify version of episerver package with a range instead of concrete version and start with lowest version that works with .net6

File: Styleguide.EPiServer/Styleguide.EPiServer.csproj [hunk 42]
```diff
@@ -1,197 +1,63 @@
...
-    <Compile Include="ContentProvider\StyleguideContentRepository.cs" />
-    <Compile Include="JsonConverters\ContentAreaConverter.cs" />
-    <Compile Include="JsonConverters\ContentConverter.cs" />
-    <Compile Include="JsonConverters\ContentReferenceConverter.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
+    <PackageReference Include="JetBrains.Annotations" Version="2022.1.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation" Version="6.0.7" />
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="6.0.0" />
+    <PackageReference Include="OdeToCode.AddFeatureFolders" Version="2.0.3" />
```
Comment (tomaszmadeyski) [inline Styleguide.EPiServer/Styleguide.EPiServer.csproj]: what's that? why do we need it?

File: Styleguide/ModuleViewCompiler.cs (hunk 32 above)
Comment (tomaszmadeyski) [reply Styleguide/ModuleViewCompiler.cs:12]: I assume this comes from here? https://stackoverflow.com/a/60901929/2393077

Can we add a comment in the code with the link to this page?

File: Styleguide/Views/Styleguide/MvcPartialComponent.cshtml (hunk 38 above)
Comment (Piotrreek) [reply Styleguide/Views/Styleguide/MvcPartialComponent.cshtml:1]: It doesn't work then. We were trying to solve that problem with Marek yesterday and only this solution worked.

File: Styleguide.EPiServer/ServiceCollectionExtensions.cs (hunk 37 above)
Comment (Piotrreek) [reply Styleguide.EPiServer/ServiceCollectionExtensions.cs:29]: I was getting DI error without it. I will test in one more time and check

**Comment by Piotrreek** [comment]
We can close this PR and move to PR with authorization
